addons/l10n_fr_invoice_addr/views/report_invoice.xml

```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <t t-set="o" t-value="o.with_context(lang=lang)"/>
+
+        <!--
+            ADDRESS BLOCK
+        -->
+
+        <!-- Remove original margin -->
+        <xpath expr="//t[@t-set='address']/address" position="attributes">
+            <attribute name="class">mb-0</attribute>
+        </xpath>
+
+        <!-- SIRET/SIREN -->
+        <xpath expr="(//t[@t-set='address'])[last()]" position="after">
+            <t t-set="address"
+               t-if="o._is_company_french() and o.partner_id.siret">
+                <t t-out="address"/>
+
+                SIRET&#160;: <t t-esc="o.partner_id.siret"/>
```

Comment:
What you could do is use a span and put the class mb-3 on it, also instead of re-setting the address, did you try to put something inside with a position inside ? 

So you could add just a span with the condition and the class